litestar/channels/base.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import AsyncGenerator, Iterable
+
+
+class ChannelsBackend(ABC):
+    @abstractmethod
+    async def on_startup(self) -> None:
+        """Called by the plugin on application startup"""
+        ...
+
+    @abstractmethod
+    async def on_shutdown(self) -> None:
+        """Called by the plugin on application shutdown"""
+        ...
+
+    @abstractmethod
+    async def publish(self, data: bytes, channels: Iterable[str]) -> None:
```

Comment:
Iterable or Sequence?

Reply:
`Iterable` is correct, doesn't need to be sequence.